pkg/solana/chainwriter/transform_registry.go

```diff
@@ -356,3 +602,48 @@ func getPoolChainConfig(sourceChainSelector, destTokenAddress []byte, poolProgra
 	poolChainConfig, _, err := solana.FindProgramAddress([][]byte{[]byte("ccip_tokenpool_chainconfig"), sourceChainSelector, destTokenAddress}, poolProgram)
 	return poolChainConfig, err
 }
+
+func ConvertToCCIPAccountMetas(metas solana.AccountMetaSlice) []ccip_offramp.CcipAccountMeta {
+	if len(metas) == 0 {
+		return nil
+	}
+	ccipMetas := make([]ccip_offramp.CcipAccountMeta, 0, len(metas))
+	for _, account := range metas {
+		ccipMetas = append(ccipMetas, ccip_offramp.CcipAccountMeta{
+			Pubkey:     account.PublicKey,
+			IsSigner:   account.IsSigner,
+			IsWritable: account.IsWritable,
+		})
+	}
+	return ccipMetas
+}
+
+func ConvertToSolanaAccountMetas(metas []ccip_offramp.CcipAccountMeta) solana.AccountMetaSlice {
+	if len(metas) == 0 {
+		return nil
+	}
+	solanaMetas := make([]*solana.AccountMeta, 0, len(metas))
+	for _, account := range metas {
+		solanaMetas = append(solanaMetas, &solana.AccountMeta{
+			PublicKey:  account.Pubkey,
+			IsSigner:   account.IsSigner,
+			IsWritable: account.IsWritable,
+		})
+	}
+	return solanaMetas
+}
+
+func encodeBigIntToFixedLengthLE(bi *big.Int, length int) []byte {
+	// Create a fixed-length byte array
+	paddedBytes := make([]byte, length)
+
+	// Use FillBytes to fill the array with big-endian data, zero-padded
+	bi.FillBytes(paddedBytes)
+
+	// Reverse the array for little-endian encoding
+	for i, j := 0, len(paddedBytes)-1; i < j; i, j = i+1, j-1 {
+		paddedBytes[i], paddedBytes[j] = paddedBytes[j], paddedBytes[i]
+	}
+
+	return paddedBytes
+}
```

Comment:
This seems ok to me, but out of curiosity: didn't we have other places where we're doing the same thing, so we could just invoke an existing function instead of creating a new one?

Reply:
Ya good eye. We use the same logic in the plugin code in the core repo but pulling this from there introduces a circular dependency. I was thinking about moving this into some utility in chainlink-common but forgot to clean it up. Think it might be better to tackle this as a follow up clean up (if we ever do) at this point in the review cycle.